ee/clickhouse/queries/trends/person.py

```diff
@@ -0,0 +1,81 @@
+from datetime import timedelta
+from typing import Dict, Tuple
+
+from dateutil.relativedelta import relativedelta
+from django.utils import timezone
+from rest_framework.utils.serializer_helpers import ReturnDict
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.models.person import ClickhousePersonSerializer
+from ee.clickhouse.queries.trends.trend_event_query import TrendsEventQuery
+from ee.clickhouse.sql.person import GET_PERSONS_FROM_EVENT_QUERY
+from posthog.constants import TRENDS_CUMULATIVE
+from posthog.models.cohort import Cohort
+from posthog.models.entity import Entity
+from posthog.models.filters import Filter
+from posthog.models.property import Property
+from posthog.models.team import Team
+
+
+def _handle_date_interval(filter: Filter) -> Filter:
+    # adhoc date handling. parsed differently with django orm
+    date_from = filter.date_from or timezone.now()
+    data: Dict = {}
+    if filter.interval == "month":
+        data.update(
+            {"date_to": (date_from + relativedelta(months=1) - timedelta(days=1)).strftime("%Y-%m-%d %H:%M:%S")}
```

Comment:
I gather from this that `date_from` and `date_to` are inclusive, hence the timedelta days=1?  What would the filter be doing in this case? Would we want to include all of "date_to" and "date_from"? and should we be setting the time component to midnight?

Out of scope but I wonder if it would make sense to use a half open interval [date_from, date_to) here instead when doing date filtering so we don't have to think about details like this.

Reply:
Ah I see this is just a copy of previous code, ignore as out of scope.

Reply:
This is one of those gnarly bits here, see also my comment to Eric. Newer code handles date ranges properly without endpoint-specific wrangling.

Your assumption for date range is kind of incorrect. Fundamentally all clickhouse date filters (assuming interval allows) end up as `timestamp >= '2021-09-10 00:00:00 AND timestamp <= '2021-09-10 23:59:59'`, but the end-of-date logic is handled separately here.